### ISSUE TITLE
[Feature request] Make alert threshold visibility in graphs optional

### ISSUE THREAD

**Issue by tomasv** [grafana/grafana#7258]
Hey 👋 

For my use-cases I don't really need to see the thresholds of an alert in my graphs and they're pretty stark visually (red and the whole alertable area is tinted as well). That's why I was thinking that perhaps it would make sense to make it optional. 

Maybe a checkbox "Show alert thresholds" in "Alert Config"? Or does "Display > Thresholds" make more sense for this?

**Comment by bergquist** [comment]
I think this makes sense. PR would be appreciated!

**Comment by tomasv** [comment]
This slipped off my radar for now, life got in the way. 😞 
I still think this would be useful, if anyone wants to pick this up then don't be blocked by me.

**Comment by dbrgn** [comment]
Here's a case where it's especially distracting:

I have a system where I collect two temperature measurements. An alert happens if the first is >50°C or if the second is >70°C.

![img](https://tmp.dbrgn.ch/screenshots/20170311215854-1oallyoi.png)

The diagram now shows a green heart to indicate that no alert situation is ongoing, but the second time series is in the red area:

![img2](https://tmp.dbrgn.ch/screenshots/20170311215930-mwx8zwiw.png)

This looks as if an alert should happen.

Being able to change the appearance or to disable the visual threshold would be greatly appreciated.

**Comment by inklit** [comment]
The alert threshold really makes my graph look like nonsense, so it would be great to make the redness optional.

Really at this point it's just polluting my graph.
http://i.imgur.com/3x9oFjG.png

**Comment by 4szt4l** [comment]
+1

Until then my workarounds:
Alert thresholds are showing only for the 1st condition.
So in the 1st line I setup something that will never happen and where the limit is outside the panel, then I add a original condition in the 2nd row with an OR.

Other alternative - if you use only 1 axis:
Move everything to the right axes and set left axis limit to something that is out of the alert range (though a (right-y) will be shown every line in the legends)

**Comment by NotTheEconomist** [comment]
Any update on this issue?

**Comment by ArmedChef** [comment]
+1 to this.  Right now I have a graph of server errors that alerts if the number of errors is greater than 0.  The whole graph is shaded red so it'd be helpful to be able to disable this display.

**Comment by Eptis** [comment]
+1 for this feature. It was fine with the threshold features

**Comment by Grygon** [comment]
+1 here too. Right now I just have a dashboard that I never look at with alerts, but that seems stupid to me when it's all just repeats of existing panels.

**Comment by HenGold** [comment]
+1 here same as @ArmedChef

**Comment by pcamelo** [comment]
+1

**Comment by CameronJHall** [comment]
+1 This would be a great feature.

**Comment by leonickel** [comment]
+1

**Comment by luchaninov** [comment]
+1

**Comment by Asafb26** [comment]
+1

**Comment by cdroot92** [comment]
+1

**Comment by pjschaafsma** [comment]
In general, the rule should be to not draw attention when no no attention is necessary, or at least allow disabling false positives (something draws attention, when actually everything is fine).

By the way, is this related to the issue where grafana keeps adding green markers to indicate alerts did not fire?

**Comment by randybb** [comment]
Probably there could be three visibility options:
- visible as we see it now,
- enabled as mark (triangle or arrow) on y axis,
- hidden.

**Comment by pfeerick** [comment]
randybb's suggestions are good. I'd also like to see this implemented, as like in dbrgn's example above I am using graphs with multiple sensors (having different ranges), so the display of the warning threshold is just adding visual pollution to the graph.

**Comment by blasco991** [comment]
+1

**Comment by aharbis** [comment]
It's also a bit weird that the thresholds are always drawn on the left axis, when the query driving the alert may be targeting the right axis. Example:

![image](https://user-images.githubusercontent.com/20443310/55686682-342b4a80-5932-11e9-8141-24c39396bf85.png)

In this graph, the alert is configured as follows:

```
WHEN avg() OF query (B, 5m, now) IS ABOVE 60
```

In this case, `B` corresponds to the right axis. However, the threshold is drawn on the left axis. Not only is the threshold and color overlay distracting, it's not even an accurate representation of the threshold for the alert.

**Comment by stieler-it** [comment]
@aharbis Maybe you should find/open another issue for that one as it is not directly related to this feature request.

**Comment by darklow** [comment]
Alerts - such a great feature, yet enabling them for all charts, makes whole dashboard look like everything is on alert – red overlays on all charts. Hope this could be fixed until end of 2025. Sorry for sarcasm, this would require single checkbox to hide overlay, yet issue is open for 2 years :/

**Comment by JulesLa** [comment]
+1

**Comment by macaba** [comment]
+1 
Just being able to turn off the fill would suit me.

**Comment by rguaitanele** [comment]
+1

**Comment by Marctraider** [comment]
+1

**Comment by Samuel29** [comment]
+1

**Comment by saxus** [comment]
+1

**Comment by lukkigi** [comment]
+1

**Comment by freedomwarrior** [comment]
+1

**Comment by sbarrier** [comment]
It would be nice to have an option like the 'toggle legend', a 'toggle alert' which remove all alert overlays (threshold and annotation) which in some case add to much noise in the graphic.

**Comment by epigns** [comment]
It will be really helpful ! +1

**Comment by RheaAyase** [comment]
Wew, **three** years for something that simple and yet so damn important. I better figure out how to hack this myself than waiting for miracle, eh?

**Comment by AlexandreGuidin** [comment]
+1

**Comment by stingalleman** [comment]
Please..!! Why is this taking so long :(

**Comment by dan-bolsun** [comment]
any updates on this?

**Comment by armageddon421** [comment]
I would really also like this. For now, I have a separate dashboard with copies of the graphs I want alerts for that I never look at. This means I don't have green hearts on the regular dashboards, but it's still better than all the clutter.

**Comment by ikkemaniac** [comment]
What about this? Seems to be working on my end. Can someone else test?

**Comment by stingalleman** [comment]
Awesome, finally! Thank you all so much!
